### ISSUE TITLE
Support reading secrets from secure files

### ISSUE THREAD

**Issue by ollykel** [spaceuptech/space-cloud#150]
Currently the config file format requires you to provide login credentials for SQL database connections like this:
```yaml
conn: "root:my-secret-pwd@/project"
```
This exposes the user's login credentials to anyone who gets access to the source code or the image.

Services like [Docker Secrets](https://docs.docker.com/engine/swarm/secrets/) avoid this by storing secrets separately from the image and providing them to the container at startup. To support this, space-cloud should provide the option to read secrets from a file. For example, instead of the conn option provided above, users should have the option to provide a "conn_file":
```yaml
conn_file: /run/secrets/mysqlconn.txt
```
Space-cloud should check that at least one option is present in the config file at startup. If both options are present it should log a fatal error.

**Comment by YourTechBud** [comment]
Thats sounds like an interesting idea! Another way we could do this is provide the secrets after space cloud has started via a configurator service. That would also help us sync the config of the various space clouds (in a Distributed environment).

But lets start by reading the secrets from a separate file. We will require a standardized format for that though.

**Comment by Jayesh333** [comment]
Would reading these secrets through environment variable be a good idea?

**Comment by YourTechBud** [comment]
Oh we can have an environment variable mentioned in the config file itself! No need for separate files in that case!

**Comment by MichaelLeeHobbs** [comment]
Is this something coming in 0.7.0? I do not even see a way to pass in the mongo creds.

**Comment by YourTechBud** [comment]
Unfortunately not. Currently you need to put the mongo credentials in the connection string itself (in the config). It will probably be in the next release.

What we are planning to do is, load the entire connection string from an environment variable.

**Comment by MichaelLeeHobbs** [comment]
Can you give an example with creds?

**Comment by YourTechBud** [comment]
Here's a link for reference:
https://docs.mongodb.com/manual/reference/connection-string/

Here's an example:
```yaml
conn: mongodb://username:secret-password@localhost:27017
```

**Comment by MichaelLeeHobbs** [comment]
@YourTechBud Thanks, sorry I even asked. I had just woke up and wasn't thinking 😊

**Comment by YourTechBud** [comment]
No problem. After implementing this feature the connection string can also be provided via an environment variable. Something like this would go in the config file:
```yaml
conn: $MY_MONGO_ENV
```

**Comment by YourTechBud** [comment]
Fixed in #217. Will release in 0.9.0
